src/main/java/com/faforever/client/leaderboard/LeaderboardController.java

```diff
@@ -0,0 +1,374 @@
+  @VisibleForTesting
+  protected AutoCompletionBinding<String> usernamesAutoCompletion;
+
+  @Override
+  public void initialize() {
+    scoreLabel.setText(i18n.get("leaderboard.score").toUpperCase());
+    searchTextField.setPromptText(i18n.get("leaderboard.searchPrompt").toUpperCase());
+
+    JavaFxUtil.bindManagedToVisible(contentPane, connectionProgressPane);
+    connectionProgressPane.visibleProperty().bind(contentPane.visibleProperty().not());
+
+    majorDivisionPicker.setConverter(divisionStringConverter());
+    yAxis.setTickLabelFormatter(integerStringConverter());
+
+    subDivisionTabPane.widthProperty().addListener((observable, oldValue, newValue) ->
+        setTabWidth((double) newValue, subDivisionTabPane.getTabs().size()));
+  }
+
+  private void setTabWidth(double tabPaneWidth, int tabNumber) {
+    // The tabs always appear 40px wider, maybe because of some obscure padding rule.
+    // We also need to subtract the tab area padding to prevent horizontal scrolling.
+    double tabWidth = Math.floor((tabPaneWidth - 20.0) / tabNumber) - 40.0;
+    JavaFxUtil.runLater(() -> {
+      subDivisionTabPane.setTabMinWidth(tabWidth);
+      subDivisionTabPane.setTabMaxWidth(tabWidth);
+    });
+  }
+
+  public void processSearchInput() {
+    String searchText = searchTextField.getText();
+    if (StringUtils.isBlank(searchText)) {
+      return;
+    }
+
+    playerService.getPlayerByName(searchText).thenAccept(playerBeanOptional -> playerBeanOptional.ifPresent(player ->
+      leaderboardService.getLeagueEntryForPlayer(player, season).thenAccept(leagueEntry -> {
+        if (leagueEntry == null) {
+          return;
+        }
+        selectLeagueEntry(leagueEntry);
+      })));
+  }
+
+  public void setSeason(LeagueSeasonBean season) {
+    this.season = season;
+
+    seasonLabel.setText(i18n.getOrDefault(
+        season.getNameKey(), String.format("leaderboard.season.%s", season.getNameKey())
+    ).toUpperCase());
+    String startDate = timeService.asDate(season.getStartDate(), FormatStyle.MEDIUM);
+    String endDate = timeService.asDate(season.getEndDate(), FormatStyle.MEDIUM);
+    seasonDateLabel.setText(i18n.get("leaderboard.seasonDate", startDate, endDate));
+    contentPane.setVisible(false);
+    searchTextField.clear();
+    if (usernamesAutoCompletion != null) {
+      usernamesAutoCompletion.dispose();
+    }
+    leaderboardService.getAllSubdivisions(season).thenAccept(this::setUsernamesAutoCompletion)
+        .exceptionally(throwable -> {
+          contentPane.setVisible(false);
+          log.warn("Error while loading division list", throwable);
+          notificationService.addImmediateErrorNotification(throwable, "leaderboard.failedToLoadDivisions");
+          return null;
+        });
+
+    playerRatingListener = observable -> updateDisplayedPlayerStats();
+    JavaFxUtil.addAndTriggerListener(playerService.getCurrentPlayer().getLeaderboardRatings(), new WeakInvalidationListener(playerRatingListener));
```

Comment:
Actually one last thing since the playerRatingListener is never changed and takes no input it can be made final and moved to the initializer, and only when the season is set do we add the listener to the leaderboardRatings